### PR TITLE
[WIP, Pulp2] Mark an importer as 'feed-updated', when changing its feed.

### PR DESCRIPTION
**Pull request by ggainey**
This allows us to re-evaluate the status of content at sync-time.

Required for #4265
https://pulp.plan.io/issues/4265

### CODE REVIEW

File: server/pulp/server/controllers/importer.py [hunk 1]
```diff
@@ -276,6 +276,8 @@ def update_importer_config(repo_id, importer_config):
     for k, v in importer_config.iteritems():
         if v is not None:
             repo_importer.config[k] = v
+            if k == importer_constants.KEY_FEED:
+                repo_importer.config[importer_constants.KEY_FEED_UPDATED] = True
```
Comment (ipanova) [inline server/pulp/server/controllers/importer.py]: @ggainey any reason we placed 'feed-updated' into the importer's config and not scratchpad?
Importer scratchpad is not editable by the user, meanwhile config yes.
```
ipanova@ina pulp]$ curl -k  -H "Content-Type:application/json" -X PUT -u admin:admin -d '{"importer_config":{"feed-updated":"data"}}' https://localhost/pulp/api/v2/repositories/zoo1/importers/yum_importer/ | python -m json.tool
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   215  100   172  100    43   1202    300 --:--:-- --:--:-- --:--:--  1503
{
    "error": null,
    "result": null,
    "spawned_tasks": [
        {
            "_href": "/pulp/api/v2/tasks/45b691eb-ff57-4633-ab0c-fe57ab1937e4/",
            "task_id": "45b691eb-ff57-4633-ab0c-fe57ab1937e4"
        }
    ]
}
[ipanova@ina pulp]$ curl -k  -H "Content-Type:application/json" -X GET -u admin:admin https://localhost/pulp/api/v2/repositories/zoo1/importers/ | python -m json.tool
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   560  100   560    0     0   6021      0 --:--:-- --:--:-- --:--:--  6021
[
    {
        "_href": "/pulp/api/v2/repositories/zoo1/importers/yum_importer/",
        "_id": {
            "$oid": "5d31d48045ef48435cb5aaf0"
        },
        "_ns": "repo_importers",
        "config": {
            "feed": "https://repos.fedorapeople.org/pulp/pulp/demo_repos/zoo1/",
            "feed-updated": "data"
        },
        "id": "yum_importer",
        "importer_type_id": "yum_importer",
        "last_override_config": {},
        "last_sync": "2019-07-19T14:33:10Z",
        "last_updated": "2019-07-22T10:02:27Z",
        "repo_id": "zoo1",
        "scratchpad": {
            "repomd_checksum": "e221dd72a3b235340d544a890432fd173bc451999b708b9c3c4b13da61ee7d9c",
            "repomd_revision": 1331832478
        }
    }
]

```

Comment (ggainey) [reply server/pulp/server/controllers/importer.py]: Yes - I didn't know 'scratchpad' exists :)  Will change.

File: server/pulp/server/db/model/__init__.py [hunk 2]
```diff
@@ -365,6 +365,34 @@ def _write_pem_file(self, config_key, path):
                 else:
                     pem_file.write(self.config[config_key])
 
+    def _set_scratchpad_entry(self, key, value):
```
Comment (ipanova) [inline server/pulp/server/db/model/__init__.py]: can we re-use already available methods? https://github.com/pulp/pulp/blob/2-master/server/pulp/plugins/conduits/mixins.py#L264
here's the example on how we work with the importer's scratchpad  https://github.com/pulp/pulp_rpm/blob/2-master/plugins/pulp_rpm/plugins/importers/yum/sync.py#L506

Comment (dkliban) [reply server/pulp/server/db/model/__init__.py]: @ggainey, @ipanova is right. We already have mixins that provide this. Sorry I led youthe wrong way.

Comment (dkliban) [reply server/pulp/server/db/model/__init__.py]: @ggainey @ipanova: the new helper methods should probably just move the mixin. The existing methods don't do enough.

Comment (ggainey) [reply server/pulp/server/db/model/__init__.py]: We have the ImporterScratchpadMixin - but at this point in the code-flow, we're dealing the Importer db-model directly, which doesn't know anything about the ImporterScratchpadMixin. That's why this code relies directly on having the helpers in the Importer model - it's all it has to work with. @ipanova @dkliban , guidance on the right way to accomplish what we want here, given the limitations of the code path?

File: server/pulp/server/controllers/importer.py (hunk 1 above)
Comment (ggainey) [reply server/pulp/server/controllers/importer.py]: Moved into scratchpad, using ImporterScratchPadMixin

File: server/pulp/server/db/model/__init__.py (hunk 2 above)
Comment (ggainey) [reply server/pulp/server/db/model/__init__.py]: Conduit is-an ImportScratchPadMixin, and available in the flows we care about - moved the helpers to ImportScratchPadMixin

**Comment by ggainey** [comment]
Closing this PR, as 4265 was closed as not-a-bug

**Pull request closed without merging**